pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -527,6 +528,19 @@ static void validateTaskConfigs(TableConfig tableConfig, Schema schema) {
                   String.format("Column \"%s\" has invalid aggregate type: %s", entry.getKey(), entry.getValue()));
             }
           }
+        } else if (taskTypeConfigName.equals(UPSERT_COMPACTION_TASK_TYPE)) {
+          // check table is realtime
+          Preconditions.checkState(tableConfig.getTableType() == TableType.REALTIME,
+              "UpsertCompactionTask only supports realtime tables!");
+          // check upsert enabled
+          Preconditions.checkState(tableConfig.isUpsertEnabled(),
+              "Upsert must be enabled for UpsertCompactionTask");
+          // check no malformed period
+          TimeUtils.convertPeriodToMillis(taskTypeConfig.getOrDefault("bufferTimePeriod", "2d"));
```

Comment:
A precondition check isn't needed because if bufferTimePeriod is malformed then `TimeUtils.convertPeriodToMillis` will raise an IllegalArgumentException.

Reply:
(minor) We don't need to do this verification if the config is not provided (no need to verify the default value)